python/ray/autoscaler/_private/node_launcher.py

```diff
@@ -6,8 +6,12 @@
 import traceback
 from typing import Any, Dict, Optional
 
+from ray.autoscaler._private.node_provider_availability_tracker import (
+    NodeProviderAvailabilityTracker,
```

Comment:
Why NodeProviderAvailabilityTracker vs. such NodeAvailabilityTracker?

Reply:
idk, there's a `node_tracker.py` that separately tracks failed nodes and it felt confusing. no strong preference though.

Reply:
mm, ok